src/nemos/utils.py

```diff
@@ -412,17 +411,45 @@ def check_invalid_entry(
     pytree_name:
         The array name.
 
-    Raises
-    ------
-        - ValueError: If any entry of `pytree` is either NaN or inf.
+    Returns
+    -------
+    is_valid:
+        A pytree with leaves flat boolean arrays of shape (n_sample_points, ) with
+        the invalid indices (samples with Nans or Infs).
```

Comment:
This has the same structure as `pytree` right? so it should say that: "Boolean pytree with same structure as `pytree`, with values specifying whether corresponding values in `pytree` are valid or not".

Reply:
The description is wrong. It returns a flat vector of boolean of length that of the sample axis. What the code does is the following:
```python
# this creates the pytree you are describing
is_valid = jax.tree_map(lambda x: check_not_inf(x) & check_not_nan(x), pytree)
# this gets the leaves of that tree and apply a logical and because we want to check all the variables
is_valid = reduce(jnp.logical_and, jax.tree_util.tree_leaves(is_valid))
```
That's the vector returned, and it should be used to filter the time axis